### DISCUSSION TITLE
how do i read the commit message in pre-commit hook

### DISCUSSION THREAD

**Discussion by pawandeore** [typicode/husky#1171] [Q&A] [answered]
hi i want to check commit message with regex pattern but the problem is 
$1  doesn’t  has commit message, so how can i get the commit message ? please anyone.. 

`

#!/usr/bin/env sh
. "$(dirname -- "$0")/_/husky.sh"

 message="$(cat $1)"

echo "MSG".$message

`

**Comment by dbowling** [comment, marked as answer]
The pre-commit hook doesn't have access to the commit message. You'll need to use the commit-msg hook to have access to the path that the temp file is saved in with the message, then you can read it.

**Comment by NielsCodes** [comment]
@dbowling I'm a little late to the party, but maybe you could point me in the right direction.
I currently have a pre-commit hook that auto fixes some linting errors. Some issues that cannot be auto-fixed, will throw an error.

This is why we would like to skip the linting script when the commit is `WIP`. Just like OP here, I was thinking of using the commit message for that, but both the `commit-msg` and `prepare-commit-msg` hooks are only executed AFTER the pre-commit hook. 
And if I add my linting script in the `commit-msg` hook, the auto fixed files are not included in my commit.

Any ideas on how to resolve this?

**Reply by dbowling** [reply]
@NielsCodes - I don't use Husky in any repos any more so I can't test this theory, but have you considered using the internal state of Git? Probably would work.

```shell
# Get the current commit message temp file
git rev-parse --git-path COMMIT_EDITMSG
# Then do something like cat and compare against your "WIP" keyword
```

**Reply by NielsCodes** [reply]
@dbowling The problem is that since it is a pre-commit hook, there is no commit yet. So the COMMIT_EDITMSG here is actually the message of my previous commit, not the one I am currently committing. But thank you for the suggestion!
